### ISSUE TITLE
Bowling Challenge: Have made a horrible mess, any help with a refactor or is it beyond saving?

### ISSUE THREAD

**Issue by MollieS** [makersacademy/issues#5]
Less of an issue and more a call for advice.  I tried to use frame objects with the bowling-challenge, and my code has ended up in a horrible mess.  My dependencies are all over the place and as for OOD, if I change anything EVERYTHING BREAKS, which I believe is the goal...

For a while it seemed as if my Frame objects could make the whole challenge much simpler, but at some point it all just got more complicated.  So, if someone wanted to have a look at my code and suggest whether I should throw it away or could suggest where it all went wrong, I would be very grateful.

https://github.com/MollieS/bowling-with-frames

**Comment by tansaku** [comment]
Hi @MollieS having a quick check I don't think that this is necessarily such a horrible mess.  In the domain of bowling game and frame are very closely interconnected.

For reference I don't think the goal of OOD is that "if I change anything EVERYTHING BREAKS".  Quite the opposite - that in a good modular OO system, changes will break subsets of things; but there is a big difference regarding the effect of changes on unit tests and on acceptance tests.

Your Frame doesn't know anything about the Game object, and so is nicely independent in that regard - so changes to the Game object will not impact the frame unit tests.

I think you could move the spare and strike score calculate into Frame if frames had knowledge about the frames coming before them.  Take a look at this concept:

http://en.wikipedia.org/wiki/Linked_list

So I don't think it's all gone wrong, but I think there is some re-arrangement of responsibilities that might allow you to slim down the Game object ...

**Comment by MollieS** [comment]
@tansaku thank you, 

Sorry, was rather unclear with my issue, I did mean that the point of OOD was to prevent the kind of issue that I am having, so I'm glad that's correct! 

I will certainly have a look at that link and I shall investigate. I wanted to keep the frame independent which is why, although I do believe it would make more sense for the frames to calculate their own scores, I was keeping the functions in the game function so as to avoid so much inter dependency, but I do like the idea of the frame of knowing its position within the game, so I will try that. 

I will keep the issue open until I've managed to refactor. 

Thank you again!
